Programs/_testembed.c

```diff
@@ -95,6 +95,14 @@ static void _testembed_Py_Initialize(void)
 }
 
 
+static int test_import_in_subinterpreters(void)
+{
+    _testembed_Py_InitializeFromConfig();
+    PyThreadState_Swap(Py_NewInterpreter());
+    return PyRun_SimpleString("import readline"); // gh-124160
+}
```

Comment:
This works, but only as long as the readline module fits the relevant criteria.  That's probably fine.

Ideally we would add a new test module for this dedicated case, though we could probably reuse _testsinglephase_with_reinit (in Modules/_testsinglephase.c).  If you think that would take much time, I'm fine with addressing that separately, so we can get this PR merged sooner.

Reply:
Would prefer if someone else figured out how to set up a more robust test case for this issue in a separate PR.